crates/universal-sierra-compiler-api/src/lib.rs

```diff
@@ -37,49 +37,44 @@ pub struct AssembledCairoProgramWithSerde {
     pub hints: Vec<(usize, Vec<Hint>)>,
 }
 
-pub fn compile_sierra_to_casm(sierra_program: &Program) -> Result<AssembledProgramWithDebugInfo> {
-    let assembled_with_info_raw = compile_sierra(
-        &serde_json::to_value(sierra_program)?,
-        None,
-        &SierraType::Raw,
-    )?;
+pub fn compile_sierra_to_casm(
+    sierra_program: &Program,
+    sierra_file_path: Option<&Utf8PathBuf>,
+) -> Result<AssembledProgramWithDebugInfo> {
```

Comment:
nit: Can we maintain same order of params as fields in `TestTargetRaw`? 
```suggestion
pub fn compile_sierra_to_casm(
    sierra_file_path: Option<&Utf8PathBuf>,
    sierra_program: &Program,
) -> Result<AssembledProgramWithDebugInfo> {
```